tests/testsuite/pub_priv.rs

```diff
@@ -198,6 +193,45 @@ Caused by:
         .run()
 }
 
+#[cargo_test]
+fn pub_dev_dependency_without_feature() {
+    Package::new("pub_dep", "0.1.0")
+        .file("src/lib.rs", "pub struct FromPub;")
+        .publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+
+                [dev-dependencies]
+                pub_dep = {version = "0.1.0", public = true}
+            "#,
+        )
+        .file(
+            "tests/mod.rs",
+            "
+            extern crate pub_dep;
+            pub fn use_pub(_: pub_dep::FromPub) {}
+        ",
+        )
+        .build();
+
+    p.cargo("check --message-format=short")
+        .masquerade_as_nightly_cargo(&["public-dependency"])
+        .with_stderr(
+            "\
+[WARNING] 'public' specifier can only be used on regular dependencies, not Development dependencies
```

Comment:
This was not introduced by this PR, but capitalized "Development dependencies" doesn't seem ideal.